### PR TITLE
fixed credentialed POST requests for predictions in dashboard (tested in notebook VM)

### PR DESCRIPTION
**Pull request by imatiach-msft**

### CODE REVIEW

File: python/interpret_community/widget/explanation_dashboard.py [hunk 1]
```diff
@@ -193,15 +243,13 @@ def __init__(self, explanation, model=None, *, dataset=None,
...
-            ExplanationDashboardInput(explanation, model, dataset, true_y, classes, features, predict_url, locale)
+        with_credentials = ExplanationDashboard.service.with_credentials
+        explanation_input = ExplanationDashboardInput(explanation, model, dataset, true_y, classes,
+                                                      features, predict_url, locale, with_credentials)
         # Due to auth, predict is only available in separate tab in cloud after login
-        if ExplanationDashboard.service.env == "local":
+        if ExplanationDashboard.service.env != "cloud":
             explanation_input.enable_predict_url()
         html = generate_inline_html(explanation_input, local_url)
-        if ExplanationDashboard.service.env == "azure":
```
Comment (rihorn2) [inline python/interpret_community/widget/explanation_dashboard.py:202]: if this now works in azure environments, the comment on line 249 can be removed :)

File: python/interpret_community/widget/js/main.js [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
-        return resp.json()
-      }
-      return Promise.reject(new Error(resp.statusText))
-    }).then(json => {
-      if (json.error !== undefined) {
-        throw new Error(json.error)
-      }
-      return Promise.resolve(json.data)
-    })
+    if (data.withCredentials) {
```
Comment (rihorn2) [inline python/interpret_community/widget/js/main.js:9]: does the axios path not handle both credintialed and non-credentialed requests? I think localhost would just ignore the headers it didn't need. Would be nice to have a single code-path if possible

File: python/interpret_community/widget/js/main.js [hunk 3]
```diff
@@ -1,23 +1,40 @@
...
+          }).then(json => {
+            if (json.error !== undefined) {
+              throw new Error(json.error)
+            }
+            return Promise.resolve(json.data)
+          })
+    }
   }
 
   ReactDOM.render(<ExplanationDashboard
```
Comment (rihorn2) [inline python/interpret_community/widget/js/main.js:40]: be sure to merge with master, don't want to revert to v1 dashboard :)

File: python/interpret_community/widget/explanation_dashboard_input.py [hunk 4]
```diff
@@ -41,7 +41,8 @@ def __init__(
             classes=None,
             features=None,
             predict_url=None,
-            locale=None):
+            locale=None,
+            with_credentials=False):
```
Comment (rihorn2) [inline python/interpret_community/widget/explanation_dashboard_input.py:45]: if the axios codepath in main.js works for uncredentailed scenarios (localhost) too, then we don't need this parameter

File: python/interpret_community/widget/js/main.js (hunk 2 above)
Comment (imatiach-msft) [reply python/interpret_community/widget/js/main.js:9]: it doesn't handle the non-credentialed request unless I remove withCredentials parameter, but then I think it's better to use the previous fetch code which is simpler and is known to work instead

File: python/interpret_community/widget/explanation_dashboard_input.py (hunk 4 above)
Comment (imatiach-msft) [reply python/interpret_community/widget/explanation_dashboard_input.py:45]: it works in local case if with_credentials is set to false - so I could use axios only but this parameter would still be needed it seems

File: python/interpret_community/widget/explanation_dashboard.py [hunk 5]
```diff
@@ -55,18 +82,66 @@ class and for the regression case a method of predict() returning the prediction
...
+                self.env = 'azure'
+            else:
+                if public_ip is not None:
+                    self.ip = public_ip
+                else:
+                    # Attempt to get the ip, but this may fail since it may not get the external ip of
+                    # the machine, just the private ip
+                    host_name = socket.gethostname()
+                    self.ip = socket.gethostbyname(host_name)
+                origin = "https://{}:{}".format(self.ip, port)
```
Comment (rihorn2) [inline python/interpret_community/widget/explanation_dashboard.py:115]: what about when port is None (the default case?)

File: python/interpret_community/widget/explanation_dashboard.py [hunk 6]
```diff
@@ -55,18 +82,66 @@ class and for the regression case a method of predict() returning the prediction
...
+                nbvm_origin2 = "https://{}-{}.{}".format(instance_name, port, domain_suffix)
+                nbvm_origins = [nbvm_origin1, nbvm_origin2]
+                headers = ['Content-Type']
+                # Support credentials for notebook VM scenario
+                self.cors = CORS(app, origins=nbvm_origins, expose_headers=headers, supports_credentials=True)
+                self.with_credentials = True
+                self.ip = LOCALHOST
+                self.env = 'azure'
+            else:
+                if public_ip is not None:
```
Comment (rihorn2) [inline python/interpret_community/widget/explanation_dashboard.py:108]: The use case of azure NB VM seems to not require with_credentials or public_ip be provided, and if they were provided, its not clear that everything would work as expected?

Comment (imatiach-msft) [reply python/interpret_community/widget/explanation_dashboard.py:108]: Azure NBVM requires with_credentials.  Local case doesn't require either parameters.  DSVM requires public ip address.  Some possible (unknown) VM like DSVM but with credentials may been both public_ip and with_credentials - I tried to setup just this (DSVM with credentials middle-layer) but ran into other issues.

File: python/interpret_community/widget/js/main.js (hunk 3 above)
Comment (imatiach-msft) [reply python/interpret_community/widget/js/main.js]: merged with master

**Pull request merged**